hive/src/test/java/com/netease/arctic/hive/catalog/HiveTableTestHelper.java

```diff
@@ -0,0 +1,118 @@
+import com.netease.arctic.data.ChangeAction;
+import com.netease.arctic.hive.io.HiveDataTestHelpers;
+import com.netease.arctic.io.DataTestHelpers;
+import com.netease.arctic.table.ArcticTable;
+import com.netease.arctic.table.KeyedTable;
+import com.netease.arctic.table.PrimaryKeySpec;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+
+public class HiveTableTestHelper extends BasicTableTestHelper {
+
+  public static final String COLUMN_NAME_OP_TIME_WITH_ZONE = "op_time_wz";
+  public static final String COLUMN_NAME_D = "d$d";
+  public static final String COLUMN_NAME_OP_DAY = "op_time_day";
+
+  public static final Schema HIVE_TABLE_SCHEMA = TypeUtil.join(
+      BasicTableTestHelper.TABLE_SCHEMA, new Schema(
+          Types.NestedField.required(5, COLUMN_NAME_OP_TIME_WITH_ZONE, Types.TimestampType.withZone()),
+          Types.NestedField.required(6, COLUMN_NAME_D, Types.DecimalType.of(10, 0)),
+          Types.NestedField.required(7, COLUMN_NAME_OP_DAY, Types.StringType.get()))
+  );
+
+  public static final PrimaryKeySpec HIVE_PRIMARY_KEY_SPEC =
+      PrimaryKeySpec.builderFor(HIVE_TABLE_SCHEMA).addColumn("id").build();
+
+  public static final PartitionSpec HIVE_SPEC = PartitionSpec.builderFor(HIVE_TABLE_SCHEMA)
+      .identity(COLUMN_NAME_OP_DAY).build();
+
+  public HiveTableTestHelper(
+      Schema tableSchema,
+      PrimaryKeySpec primaryKeySpec,
+      PartitionSpec partitionSpec,
+      Map<String, String> tableProperties) {
+    super(tableSchema, primaryKeySpec, partitionSpec, tableProperties);
+  }
+
+  public HiveTableTestHelper(
+      boolean hasPrimaryKey,
+      boolean hasPartition,
+      Map<String, String> tableProperties) {
+    this(HIVE_TABLE_SCHEMA, hasPrimaryKey ? HIVE_PRIMARY_KEY_SPEC : PrimaryKeySpec.noPrimaryKey(),
+        hasPartition ? HIVE_SPEC : PartitionSpec.unpartitioned(), tableProperties);
+  }
+
+  public HiveTableTestHelper(boolean hasPrimaryKey, boolean hasPartition) {
+    this(hasPrimaryKey, hasPartition, Maps.newHashMap());
+  }
+
+  @Override
+  public Record generateTestRecord(int id, String name, long ts, String opTime) {
+    return DataTestHelpers.createRecord(HIVE_TABLE_SCHEMA, id, name, ts, opTime,
+        opTime + "Z", new BigDecimal("0"), opTime.substring(0, 10));
+  }
+
+  @Override
+  public List<DataFile> writeChangeStore(
+      KeyedTable keyedTable, long txId, ChangeAction action, List<Record> records, boolean orderedWrite) {
+    return HiveDataTestHelpers.writeChangeStore(keyedTable, txId, action, records, orderedWrite);
+  }
+
+  @Override
+  public List<DataFile> writeBaseStore(
```

Comment:
Do we need `writeHiveStore` method?

Reply:
`TableTestHelper` abstracted the common operations on Arctic tables.
Use `HiveDataTestHelpers#writeBaseStore` to write files in the Hive location.

Reply:
1